sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeActionsOperation.cs

```diff
@@ -152,7 +171,10 @@ internal AnalyzeActionsOperation(TextAnalyticsRestClient serviceClient, ClientDi
 
             // TODO: Add validation here
             // https://github.com/Azure/azure-sdk-for-net/issues/11505
-            Id = operationLocation.Split('/').Last();
+            _jobId = operationLocation.Split('/').Last();
+
+            string plainId = JsonSerializer.Serialize(new OperationIdInformation(_jobId, idToIndexMap, showStats));
+            Id = ClientCommon.EncodeOperationId(plainId);
```

Comment:
Ditto on possibly moving this into `OperationIdInformation` so all the logic is grouped in one spot.